### ISSUE TITLE
Extend our study to models trained on langugae

### ISSUE THREAD

**Issue by msakarvadia** [msakarvadia/memorization#15]
Think about how to extend our study to include natural language.

One option is to train toy models on a small data set; maybe [shakespeare](https://huggingface.co/datasets/Trelis/tiny-shakespeare). Induce errors like word swaps or spelling mistakes and quantify how much is memorized.

Apply memorization mitigation strategies.

- [x] Try training on Shakespeare data
- [x] Try training on Wikipedia data
- [x] random noise,
- [x] backdoors
- [x] Need to make the tokenization process faster
- [x] Need to pick triggers more systematically
- [x] duplicates - generate during training
- [x] Need to measure memorization in each of the above three settings w/ duplicates (ASR for backdoor, % mem for noise)
- [x] fix pad, eos, bos tokens when initializing model  + when generating come up with attention mask (maybe make fake attn mask).
- [x] fix pad token for refined_check_perc_mem func
- [x] Data should return clean version of backdoored test set as well
- [x] Maybe only measure memorization every X epochs due to computational cost -  might need to through out previous training ckpts
